images/virtualization-artifact/pkg/controller/vm/internal/util.go

```diff
@@ -159,3 +159,9 @@ func isInternalVirtualMachineError(phase virtv1.VirtualMachinePrintableStatus) b
 func podFinal(pod corev1.Pod) bool {
 	return pod.Status.Phase == corev1.PodSucceeded || pod.Status.Phase == corev1.PodFailed
 }
+
+func newSetCondition(vm *virtv2.VirtualMachine) func(c conditions.Conder) {
```

Comment:
Don't see the reason to wrap here
Just use `conditions.SetCondition(c, &vm.Status.Conditions)` and keep it simple

Reply:
ok